### PR TITLE
Bug in stake using stash address in polkadot

### LINKED ISSUE #4142: Bug in polkadot staking using stash address
**Describe the bug**

- When I want to stake using a stash address I can continue even if the amount is 0. This is wrong because the only allowed action is to increase stake, so that doesn't make sense.
- If I  enter a valid amount I get the error `Variable "$message" got invalid value { to: ["GTzRQPzkcuynHgkEHhsPBFpKdh4sAacVRsnd8vYfPpTMeEY"], amount: { amount: "0.01", denom: "KSM" }, addressRole: "stash" }; Field "addressRole" is not defined by type "TransactionDetailsInput".`

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: 
 - Browser: 
 - Version: 

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser 
 - Version 

**Additional context**

### PR DESCRIPTION
**Pull request by mariopino**
Closes #4142

**Description:**

Amount should always > 0 at staking using a stash address in polkadot,

Thank you! 🚀

---

For contributor:

- [x] Added changes entries. Run `yarn changelog` for a guided process.
- [x] Reviewed `Files changed` in the github PR explorer
- [x] Attach screenshots of the UI components on the PR description (if applicable)
- [x] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -965,6 +964,7 @@ export default {
       },
       /* istanbul ignore next */
       update(data) {
+        console.log(data)
```
Comment (Bitcoinera) [inline src/ActionModal/components/ActionModal.vue:967]: Please remove 🙏

File: src/ActionModal/components/ActionModal.vue [hunk 2]
```diff
@@ -942,7 +942,6 @@ export default {
       variables() {
         let { type, ...message } = this.transactionData
         delete message.memo
-        delete message.addressRole
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue:945]: why the delete?

File: src/ActionModal/components/DelegationModal.vue [hunk 3]
```diff
@@ -358,17 +358,7 @@ export default {
...
-            (this.currentNetwork.network_type === "polkadot" &&
-              this.totalStaked > 0) ||
-            this.session.addressRole === `controller`
-          ) {
-            return true
-          } else {
-            return !!x && x !== `0`
-          }
-        },
+        required: (x) => (!!x && x !== `0`) || this.session.addressRole === `controller`,
```
Comment (faboweb) [inline src/ActionModal/components/DelegationModal.vue:361]: nice simplification, but could be confusing in the future

File: src/ActionModal/components/ActionModal.vue (hunk 2 above)
Comment (mariopino) [reply src/ActionModal/components/ActionModal.vue:945]: We need to pass addressRole to API to be able to calculate the fee in a stakeTx.

File: src/ActionModal/components/DelegationModal.vue (hunk 3 above)
Comment (mariopino) [reply src/ActionModal/components/DelegationModal.vue:361]: Why? Can we improve somehow?

Comment (mariopino) [reply src/ActionModal/components/DelegationModal.vue:361]: BTW: I tested using stash, controller and stash/controller in polkadot and stake/restake in cosmos just to make sure it works ^^

Comment (faboweb) [reply src/ActionModal/components/DelegationModal.vue:361]: just mentioning, because addressRole is somehow implicit to polkadot.

Comment (mariopino) [reply src/ActionModal/components/DelegationModal.vue:361]: Yep, true! the good thing is that we can safely skip it for other network types (I hope ^^)

**Pull request merged**